### DISCUSSION TITLE
PI_RF/FM_RF

### DISCUSSION THREAD

**Discussion by MikeMcPherson** [alanbjohnston/CubeSatSim#333] [General]
I've studied the repo and the schematic, and I'll be damned if I can figure out why there are two antennas on the new board! :-)  What am I missing?

Thanks.  Mike

**Comment by alanbjohnston** [comment]
Hi Mike,

If you look at the block diagram from the wiki ![CubeSatSim Block Diagram](https://github.com/alanbjohnston/CubeSatSim/assets/25288070/eaa18480-74d0-492d-8086-38b1c7077113), one antenna is connected to the SR105U FM transceiver module, while the other is connected to the GPIO clock pin on the Raspberry Pi Zero.

The original v1 CubeSatSim just had one transmitter, the GPIO clock pin on the Pi Zero using the librpitx library https://github.com/F5OEO/librpitx.  The Beta v1.3 design has that original transmitter plus the SR105U FM transceiver board.

Having two antennas allows the two radios to be run at the same time.  We use this when we are transmitting in FSK or BPSK modes with one antenna and receiving at the same time with the SR105U FM transceiver with the other antenna for Radio Command and Control signals. (See the new Command and Control capabilities that can be added to any Beta CubeSatSim: https://github.com/alanbjohnston/CubeSatSim/wiki/Command-and-Control

Finally, by cutting JP3 and bridging JP7, both transmitters can use just one antenna.  This works surprisingly well, but having separate antennas is better.  If the 20dB attenuator circuit weren't present, the FM transceiver would likely damage the Pi Zero GPIO clock pin, but with this circuit in place, I haven't had any issues when I have tested this.

73,
Alan

**Comment by Luna94-maker** [comment]
Dual antennas on a board often serve specific purposes, such as diversity reception (switching between antennas to combat signal fading) or supporting multiple frequency bands. Another possibility is that one is a dedicated GPS/GNSS antenna, while the other handles primary communication. For testing setups, you might even see configurations that use an RF Attenuator between them to simulate path loss. If the schematic shows components like splitters or switches nearby, that would confirm this theory. The team at Zomwave offers various [RF Attenuator](https://zomwave.com/product-category/coaxial-rf-attenuator/) options that could help in testing such multi-antenna systems. Have you checked if the antennas are labeled with different frequency specifications?

**Reply by alanbjohnston** [reply]
Yes, these are all valid reasons for having multiple antennas on a CubeSat. 

In the CubeSatSim, both antennas are for the 70cm/ISM band.

When the CubeSatSim is operating in Mode 6 Cross Band Repeater, the rpitx transmitter is actually operating on the 2m band, so a 2m antenna could be used.

If the Low Pass Filters on each were changed, a different transmit and receive frequency could be used. 

Alan
